### ISSUE TITLE
Regresion - Application crash when non ascii character in home path

### ISSUE THREAD

**Issue by imkebe** [Tribler/tribler#1987]
My username contains character "ł". Homepath C:\Users\Sławomir
6.4.3 version worked.

- [ x ] I have read CONTRIBUTING.md
- [ x ] I have tried with the latest pre-release version and I still can reproduce the issue.

##### Tribler version/branch+revision: #####
6.5.0 
##### Operating system and version: #####
Windows 10 
##### Expected behavior: #####
Up running
##### Actual behavior: #####
Crash
##### Steps to reproduce the behavior: #####
Run app
##### Relevant log file output: #####
Traceback (most recent call last):
  File "Tribler\Main\tribler_main.pyo", line 1070, in run
  File "Tribler\Main\tribler_main.pyo", line 147, in __init__
  File "Tribler\Main\tribler_main.pyo", line 314, in InitStage1
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb3 in position 2397: ordinal not in range(128)

**Comment by whirm** [comment]
Hm, that's strange, I can run Tribler 6.5 on a windows 10 machine as the user `四国领导人`...

I'll have a look at it on Monday.

Thanks for the report!

**Comment by LipuFei** [comment]
@whirm Try to use the Win32 API like `getEnvironmentVariable()` in tribler_exe.py.

**Comment by whirm** [comment]
@LipuFei that's what we are doing already: https://github.com/Tribler/tribler/blob/be52821a6384eb861427c2964a9a7060450df3ef/Tribler/Main/tribler_exe.py#L53

**Comment by LipuFei** [comment]
Check tribler_main.py line 992:
```
installdir = determine_install_dir()
```

`determine_install_dir()` is not using Win32 API. Could be a problem.

**Comment by whirm** [comment]
@imkebe where is Tribler installed?

**Comment by LipuFei** [comment]
@whirm I remember from the last time that the installation path must be normal ascii, because otherwise our NSIS installer will not work. But I also remember something strange with this: I think if you have Chinese characters NSIS won't install, but if you have European language unicode characters it will work.

**Comment by whirm** [comment]
@LipuFei Hm, I thought it was just not compatible at all with non-ascii paths. Let's wait to see where @imkebe installed it to make sure that's the actual issue.

**Comment by imkebe** [comment]
@whirm default path
C:\Program Files (x86)\Tribler\

**Comment by whirm** [comment]
@imkebe I've created a `Sławomir` user on my testing machine and Tribler is running just fine there.

Could you upload your `tribler.conf` and `libtribler.conf` files? (they are located in `%APPDATA%/.Tribler`I suspect some path on the config file which I guess you carried from 6.4 is not being decoded properly.

**Comment by imkebe** [comment]
[tribler.conf](https://github.com/Tribler/tribler/files/156797/tribler.conf.txt)
[libtribler.conf](https://github.com/Tribler/tribler/files/156798/libtribler.conf.txt)

**Comment by imkebe** [comment]
There is something in user path. I deleted entire %APPDATA%/.Tribler directory and it isn't recreated on Tribler start. It just stucks at the start.

---------------------------
Errors occurred
---------------------------
See the logfile 'u'C:\\Users\\S\u0142awomir\\AppData\\Roaming\\tribler.exe.log'' for details
---------------------------
OK   
---------------------------

**Comment by whirm** [comment]
does tribler.exe.log contain anything else than the stracktrace you already pasted?

**Comment by whirm** [comment]
Could you paste this:
```
def determine_install_dir():
    # Niels, 2011-03-03: Working dir sometimes set to a browsers working dir
    # only seen on windows

    # apply trick to obtain the executable location
    # see http://www.py2exe.org/index.cgi/WhereAmI
    # Niels, 2012-01-31: py2exe should only apply to windows

    # TODO(emilon): tribler_main.py is not frozen, so I think the special
    # treatment for windows could be removed (Needs to be tested)
    if sys.platform == 'win32':
        def we_are_frozen():
            """Returns whether we are frozen via py2exe.
            This will affect how we find out where we are located."""
            return hasattr(sys, "frozen")

        def module_path():
            """ This will get us the program's directory,
            even if we are frozen using py2exe"""
            if we_are_frozen():
                inst_dir = os.path.dirname(unicode(sys.executable, sys.getfilesystemencoding()))
                print >> sys.stderr, "WE ARE FROZEN", repr(inst_dir)
                return inst_dir

            filedir = os.path.dirname(unicode(__file__, sys.getfilesystemencoding()))
            inst_dir =  os.path.abspath(os.path.join(filedir, '..', '..', '..'))
            print >> sys.stderr, "INST DIR IS", repr(inst_dir)
            return inst_dir

        return module_path()
    elif sys.platform == 'darwin':
        # On a packaged app, this file will be at:
        # Tribler.app/Contents/Resources/lib/Python2.7/site-packages.zip/Tribler/Core/Utilities/install_dir.py
        cur_file = os.path.dirname(__file__)
        if "site-packages.zip" in cur_file:
            return os.path.abspath(os.path.join(cur_file, '..', '..', '..', '..', '..', '..'))
        # Otherwise do the same than on Unix/Linux

    this_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
    return '/usr/share/tribler' if this_dir.startswith('/usr/lib') else this_dir
```

into a file called `install_dir.py` and place it in `C:\Program Files (x86)\Tribler\Tribler\Core\Utilities\`
And then rename the `install_dir.pyo` located in the same dir to something else?
That should print some useful info on the log file.

Thanks!

**Comment by imkebe** [comment]
```
Traceback (most recent call last):
  File "tribler_exe.py", line 80, in <module>
  File "Tribler\Main\tribler.pyo", line 9, in <module>
    
  File "C:\Program Files (x86)\Tribler\.\Tribler\Core\Utilities\install_dir.py", line 11, in determine_install_dir
    if sys.platform == 'win32':
NameError: global name 'sys' is not defined
```

**Comment by whirm** [comment]
my bad, add `import sys` as the first line of the file you've created and try again.

**Comment by imkebe** [comment]
You forget about `import os` also :) But that gives no new data. 

```
WE ARE FROZEN u'C:\\Program Files (x86)\\Tribler'
WE ARE FROZEN u'C:\\Program Files (x86)\\Tribler'
Traceback (most recent call last):
  File "Tribler\Main\tribler_main.pyo", line 1070, in run
  File "Tribler\Main\tribler_main.pyo", line 147, in __init__
  File "Tribler\Main\tribler_main.pyo", line 314, in InitStage1
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb3 in position 2397: ordinal not in range(128)
```

Maybe that will help ? http://www.azavea.com/blogs/labs/2014/03/solving-unicode-problems-in-python-2-7/

**Comment by whirm** [comment]
No, I know what's going on now. Apparently your `PATH` contains non-ASCII paths and Tribler is updating it's own copy of the variable, this breaks the normal Python API. 

I'll push a fix for that later today. 

Thanks for the help!

**Comment by whirm** [comment]
@imkebe I've merged a fix for the non-ASCII `%PATH%` and another for the repeated log line. I've tested it on Windows 10 by adding a non-ASCII path to my `%PATH%`.

Could you see if installing this build fixes it for you?
https://jenkins.tribler.org/job/Publish_tribler_next/193/

**Comment by imkebe** [comment]
Issue still exist but there are new log data:
```
Traceback (most recent call last):
  File "Tribler\Main\tribler_main.pyo", line 1067, in run
  File "Tribler\Main\tribler_main.pyo", line 147, in __init__
  File "Tribler\Main\tribler_main.pyo", line 384, in InitStage1
  File "Tribler\Core\Session.pyo", line 71, in __init__
  File "Tribler\Core\SessionConfig.pyo", line 69, in __init__
  File "distutils\spawn.pyo", line 220, in find_executable
  File "ntpath.pyo", line 85, in join
UnicodeDecodeError: 'ascii' codec can't decode byte 0xb3 in position 8: ordinal not in range(128)
```

**Comment by whirm** [comment]
I'll fix that one ASAP, thx!

**Comment by whirm** [comment]
@imkebe could you give a try to this build? 

https://jenkins.tribler.org/job/Build-Tribler_WinSrvr2008R2_64_emilon/169/

**Comment by imkebe** [comment]
It works.

**Comment by whirm** [comment]
Great! I'll merge that then, I want to fix another bug and I'll release V6.5.1

Thanks!
